### ISSUE TITLE
TypeScript issues due to exported const enums

### ISSUE THREAD

**Issue by NateRickard** [microsoft/appcenter-sdk-react-native#788]
# **Description**

When using TypeScript to write a React Native app (with the App Center SDK(s) added), many of the enum types (e.g. `UserConfirmation` from the Crashes SDK) are unusable due to improperly exporting them from the JS module(s). Any `enum` exported with the `export const enum` syntax will not be accessible. When attempting to use one of these enum types, TypeScript shows the following compilation error:

>Cannot access ambient const enums when the '--isolatedModules' flag is provided. ts(2748)

The root cause and further discussion on the background of this issue [can be found here](https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats) and [here](https://github.com/facebook/create-react-app/issues/5738). To resolve the issue, the enum can be exported without the `const` keyword.

This issue is encountered with the default TypeScript settings shipped with the latest React Native project template. Note `UserConfirmation` is an example, and this issue occurs with **all** JavaScript enums exported with the `export const enum` syntax.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Configure a React Native project that uses TypeScript
2. Add the App Center Crashes SDK and attempt to use the `UserConfirmation` enum
3. TypeScript will "red squiggly underline" the enum usage with the error (if using VSCode)

### **Details**

1. Which version of the App Center SDK are you using?
    - 2.5.0
2. Which OS version did you experience the issue on?
    - All - issue occurs pre-compilation
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Any/None - issue occurs pre-compilation
4. What third party libraries are you using?
    - n/a
5. Run the following command and paste the output below: `react-native info`

```
warn The following packages use deprecated "rnpm" config that will stop working from next release:
  - react-native-code-push: https://microsoft.github.io/code-push
Please notify their maintainers about it. You can find more details at https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide.
info Fetching system and libraries information...
System:
    OS: macOS 10.15.2
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 3.82 GB / 32.00 GB
    Shell: 5.0.11 - /usr/local/bin/bash
  Binaries:
    Node: 12.12.0 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-25 | Google Play Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom
      Android NDK: 20.1.5948944
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5977832
    Xcode: 11.3/11C29 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.3 => 0.61.3
```

6. If you're developing for React Native iOS, run the following command and paste the output below: `pod --version`

`1.8.4`

7. Please enable verbose logging for your app using `[MSAppCenter setLogLevel: MSLogLevelVerbose];` before your call to `[AppCenterReactNative register];` (or any other SDK registration). For Android, use `AppCenter.setLogLevel(android.util.Log.VERBOSE);` before your `SoLoader.init` call (you will need to `import com.microsoft.appcenter.AppCenter;`). Include the logs here:

N/A - not a runtime issue and/or version specific

**Comment by guperrot** [comment]
Hi, just to confirm we would need only to remove `const` from enums and nothing else?

**Comment by guperrot** [comment]
This bug was already fixed in version [2.6.0](https://github.com/microsoft/appcenter-sdk-react-native/releases/tag/2.6.0) of the SDK: https://github.com/microsoft/appcenter-sdk-react-native/pull/767/files.

The latest version of the SDK is [2.6.1](https://github.com/microsoft/appcenter-sdk-react-native/releases/tag/2.6.1).
